compiler/rustc_data_structures/src/sync.rs

```diff
@@ -30,6 +31,35 @@ pub use vec::AppendOnlyVec;
 
 mod vec;
 
+mod mode {
+    use super::Ordering;
+    use std::sync::atomic::AtomicU8;
+
+    const UNINITIALIZED: u8 = 0;
+    const SINGLE_THREAD: u8 = 1;
+    const MULTIPLE_THREADS: u8 = 2;
```

Comment:
Naming these `ACTIVE` and `INACTIVE` would better match the `active` function name.

Reply:
Done
In this way, should we make `INACTIVE` the default choice when the user does not explicitly set the mode? This way it looks more like a switch for users.

Reply:
It probably makes sense to be explicit here.